packages/databases-collections/src/components/collection-fields/collection-fields.jsx

```diff
@@ -0,0 +1,313 @@
+    onToggle: PropTypes.func.isRequired,
+    toggled: PropTypes.bool,
+    openLink: PropTypes.func
+  }
+
+  render() {
+    return (
+      <FieldSet>
+        <Checkbox
+          onChange={event => {
+            this.props.onToggle(event.target.checked);
+          }}
+          label={this.props.label}
+          checked={this.props.toggled}
+          bold={false}
+        />
+        {!this.props.description ? '' : this.props.description}
+        {!!this.props.helpUrl && (
+          <IconButton
+            className={styles['info-btn']}
+            aria-label="Time-series collections documentation"
+            onClick={() => this.props.openLink(this.props.helpUrl)}
+          >
+            <Icon
+              glyph="InfoWithCircle"
+              size="small"
+            />
+          </IconButton>
+        )}
+        {!this.props.toggled ? '' : this.props.children}
+      </FieldSet>
+    );
+  }
+}
+
+export default class CollectionFields extends PureComponent {
+  static propTypes = {
+    onChange: PropTypes.func.isRequired,
+    withDatabase: PropTypes.bool,
+    serverVersion: PropTypes.string,
+    openLink: PropTypes.func.isRequired
+  }
+
+  constructor() {
+    super();
+    this.state = {
+      fields: {}
+    };
+  }
+
+  setField(fieldName, value) {
+    const fields = _.cloneDeep(this.state.fields);
+    _.set(fields, fieldName, Object.is(value, NaN) ? undefined : value);
+    this.setState({ fields }, this.updateOptions);
+  }
+
+  setCollationField(fieldName, value) {
+    // We convert the value to a string for the option value.
+    // Here we reset it to its original type.
+    if (!Object.is(this.asNumber(value), NaN)) {
+      this.setField(`collation.${fieldName}`, this.asNumber(value));
+      return;
+    }
+    if (value === 'false') {
+      this.setField(`collation.${fieldName}`, false);
+      return;
+    }
+    if (value === 'true') {
+      this.setField(`collation.${fieldName}`, true);
+      return;
+    }
+
+    this.setField(`collation.${fieldName}`, value);
+  }
+
+  updateOptions = () => {
+    this.props.onChange({
+      database: this.state.fields.databaseName,
+      collection: this.state.fields.collectionName,
+      options: this.buildOptions()
+    });
+  }
+
+  asNumber(value) {
+    return `${value}` ? +value : undefined;
+  }
+
+  buildOptions() {
+    const state = this.state;
+    const cappedOptions = state.isCapped ? {
+      capped: true,
+      size: state.fields.cappedSize ? state.fields.cappedSize : undefined
+    } : {};
+
+    const collationOptions = state.isCustomCollation ? {
+      collation: state.fields.collation
+    } : {};
+
+    const timeSeriesOptions = state.isTimeSeries ? {
+      timeseries: state.fields.timeSeries
+    } : {};
+
+    return {
+      ...collationOptions,
+      ...cappedOptions,
+      ...timeSeriesOptions
+    };
+  }
+
+  renderDatabaseNameField() {
+    return (<FieldSet>
+      <TextInput
+        autoFocus
```

Comment:
Nice catch.  I think this is a bit of a special case because these fields are in a modal. I think we do want the modal to auto focus the content of the modal so someone using a motor control or pure keyboard can interact with the modal content immediately instead of having to navigate somewhere. I think the modal component does this for us though and we shouldn't have these inputs trying to do that functionality. removed 👍

Reply:
Yeah, leafygreen seems to correctly lock focus inside the modal 👍